packages/catalog-process/src/model/domain/errors.ts

```diff
@@ -354,3 +355,13 @@ export function eServiceTemplateWithoutPublishedVersion(
     title: "EService template without published version",
   });
 }
+
+export function templateIdMustBeUndefined(
```

Comment:
I would change the naming of the assertion to something like `assertEServiceNotTemplateInstance`